[BE-3747] feat: SSO mapping documentation (#527)
appcircleio/appcircle-docusaurus: docs/account/my-organization/sso-providers-configuration/auth-saml.md

When you connect your Identity Provider, please open a new incognito window and test the SSO integration. Please only log off when you can log in with SSO credentials. If the connection doesn't work, you need to review your settings.

:::

### SSO Mapping

> This step is optional. You can skip it if you do not plan to use SSO Mapping.

Reviewer: ```suggestion
This step is optional and can be skipped if you do not plan to use SSO Mapping.
```

- Navigate to the **User Management** section in the Auth0 Dashboard, click on **Roles**, and create the roles as needed.

<Screenshot url='https://cdn.appcircle.io/docs/assets/sso-mapping-auth0-create-roles.png' />

- Navigate to the organization section and create organizations.

<Screenshot url='https://cdn.appcircle.io/docs/assets/sso-mapping-auth0-create-groups.png' />

> - Add users who will become members of your organization

Reviewer: ```suggestion
- Add users who will become members of your organization.
```

<Screenshot url='https://cdn.appcircle.io/docs/assets/sso-mapping-auth0-add-members.png' />

- Click on the three dots and select **Assign Roles**. Assign the desired roles to users.

<Screenshot url='https://cdn.appcircle.io/docs/assets/sso-mapping-auth0-assign-roles.png' />

> - Navigate to the **Connections** tab. Enable Connections for your organization. 

Reviewer: ```suggestion
- Navigate to the **Connections** tab and enable Connections for your organization. 
```

<Screenshot url='https://cdn.appcircle.io/docs/assets/sso-mapping-auth0-enable-connections.png' />

> - Navigate to the Applications section. Go to the related application and then navigate to the Organizations tab. Click on **Disable Grants Now** 

Reviewer: ```suggestion
- To create roles, navigate to the **User Management** section in the Auth0 Dashboard, click on **"Roles"**, and create the necessary roles.
```

> - Navigate to the Applications section. Go to the related application and then navigate to the Organizations tab. Click on **Disable Grants Now** 

Reviewer: ```suggestion
- Navigate to the "Applications" section. Select the relevant application, then go to the "Organizations" tab. Click on **"Disable Grants Now."**
```

<Screenshot url='https://cdn.appcircle.io/docs/assets/sso-mapping-auth0-application-organizations1.png' />

> - Choose **Business** Users for the type of users and select **Prompt for Organization** for the Login Flow.)Click on **Save Changes**

Reviewer: ```suggestion
Choose "**Business Users**" for the type of users and select "**Prompt for Organization**" for the login flow. Click on "**Save Changes**."
```

<Screenshot url='https://cdn.appcircle.io/docs/assets/sso-mapping-auth0-application-organizations2.png' />

> - We need to retrieve the groups and roles of a user via tokens. The groups claim is already present on the token, but we need to add an additional step for the roles claim. Navigate to **Actions** and click on the **Library** tab. Click on the **Create Action** and select the **Build from scratch**

Reviewer: ```suggestion
- To retrieve the groups and roles of a user via tokens, follow these steps:

1. Navigate to **Actions**.
2. Click on the **Library** tab.
3. Select **Create Action**.
4. Choose **Build from Scratch**.

The groups claim is already present on the token, but these steps will help you add the roles claim as well.
```

<Screenshot url='https://cdn.appcircle.io/docs/assets/sso-mapping-auth0-actions-library1.png' />

- Give an appropriate name to the custom action. 

<Screenshot url='https://cdn.appcircle.io/docs/assets/sso-mapping-auth0-actions-library2.png' />

- Paste following Javascript code and click on the deploy.

<Screenshot url='https://cdn.appcircle.io/docs/assets/sso-mapping-auth0-actions-library3.png' />

```js
exports.onExecutePostLogin = async (event, api) => {
  const namespace = 'your_namespace';
  if (event.authorization) {
    api.idToken.setCustomClaim(`${namespace}roles`, event.authorization.roles);
    api.accessToken.setCustomClaim(`${namespace}roles`, event.authorization.roles);
  }
}
```
- Navigate to **Actions** and click on the **Flows** tab. 
- Click on the **Login**. 

<Screenshot url='https://cdn.appcircle.io/docs/assets/sso-mapping-auth0-actions-flows1.png' /> 

- Drag and drop the custom action created previously. 

<Screenshot url='https://cdn.appcircle.io/docs/assets/sso-mapping-auth0-actions-flows2.png' />

> - Go back to Appcircle, enter **Group Attribute Name** as ``http://schemas.auth0.com/org_id`` and **Role Attribute Name** as ``http://schemas.auth0.com/your_namespace_roles``. Note that the role claim is created as a custom claim in Auth0. You must enter the name you determined before.

Reviewer: ```suggestion
- Return to Appcircle, enter **Group Attribute Name** as ``http://schemas.auth0.com/org_id`` and **Role Attribute Name** as ``http://schemas.auth0.com/your_namespace_roles``. Note that the role claim is created as a custom claim in Auth0 so you must enter the name you determined previously.
```

:::

### SSO Mapping

This step is optional and can be skipped if you do not plan to use SSO Mapping.

- Navigate to the **User Management** section in the Auth0 Dashboard, click on **Roles**, and create the roles as needed.

<Screenshot url='https://cdn.appcircle.io/docs/assets/sso-mapping-auth0-create-roles.png' />

- Navigate to the organization section and create organizations.

<Screenshot url='https://cdn.appcircle.io/docs/assets/sso-mapping-auth0-create-groups.png' />

- Add users who will become members of your organization.

<Screenshot url='https://cdn.appcircle.io/docs/assets/sso-mapping-auth0-add-members.png' />

- Click on the three dots and select **Assign Roles**. Assign the desired roles to users.

<Screenshot url='https://cdn.appcircle.io/docs/assets/sso-mapping-auth0-assign-roles.png' />

- Navigate to the **Connections** tab and enable Connections for your organization. 

<Screenshot url='https://cdn.appcircle.io/docs/assets/sso-mapping-auth0-enable-connections.png' />

> - Navigate to the "Applications" section. Select the relevant application, then go to the "Organizations" tab. Click on **"Disable Grants Now."**

Reviewer: ```suggestion
- Navigate to the "Applications" section. Select the relevant application, then go to the "Organizations" tab. Click on **"Disable Grants Now"**.
```

<Screenshot url='https://cdn.appcircle.io/docs/assets/sso-mapping-auth0-application-organizations1.png' />

> Choose "**Business Users**" for the type of users and select "**Prompt for Organization**" for the login flow. Click on "**Save Changes**."

Reviewer: ```suggestion
Choose "**Business Users**" for the type of users and select "**Prompt for Organization**" for the login flow. Click on "**Save Changes**".
```
